l2/main.tf

```diff
@@ -8,3 +8,9 @@ module "alb" {
   instance_id = module.ec2.instance_id
 
 }
+
+module "asg" {
+  source   = "../modules/asg"
+  image_id = "ami-0359b3157f016ae46"
```

Comment:
Please use aws_ami datasource instead of hardcoding image_id:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ami

Reply:
updated